service/storage_service.cc

```diff
@@ -6936,6 +6936,61 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
     });
 }
 
+future<locator::load_stats> storage_service::load_stats_for_tablet_based_tables() {
+    auto holder = _async_gate.hold();
+
+    if (this_shard_id() != 0) {
+        // group0 is only set on shard 0.
+        co_return co_await container().invoke_on(0, [&] (auto& ss) {
+            return ss.load_stats_for_tablet_based_tables();
+        });
+    }
+
+    auto load_stats = co_await _db.map_reduce0([] (replica::database& db) mutable -> future<locator::load_stats> {
+        locator::load_stats load_stats{};
+        auto& tables_metadata = db.get_tables_metadata();
+
+        co_await tables_metadata.for_each_table_gently([&] (table_id id, lw_shared_ptr<replica::table> table) -> future<> {
```

Comment:
It's wasteful to iterate on every table. If we have 1000 tables and 100 nodes, we'll send 100,000 RPCs.

Better would be to ask each node for its map, and update a global map here. If a tablet is migrating, its data will be out-of-date, but that's no big deal.

Anyway it can be optimized later.

Reply:
This is the RPC handler. Each node combines a tablet map with info from each shards. So in your example, it would be 100 RPCs, with each node building a map from the 1k tables and returning it to the coordinator.

Reply:
ok